MemoryMeter.c

```diff
@@ -19,17 +19,17 @@ static const int MemoryMeter_attributes[] = {
    MEMORY_CACHE
 };
 
-static void MemoryMeter_updateValues(Meter* this, char* buffer, int size) {
+static void MemoryMeter_updateValues(Meter* this, char* buffer, size_t size) {
    int written;
    Platform_setMemoryValues(this);
 
    written = Meter_humanUnit(buffer, this->values[0], size);
-   buffer += written;
-   if ((size -= written) > 0) {
-      *buffer++ = '/';
-      size--;
-      Meter_humanUnit(buffer, this->total, size);
-   }
+   METER_BUFFER_CHECK(buffer, size, written);
+
+   *buffer = '/';
```

Comment:
Should check for `size>1` and add `NUL`-termination. Otherwise may leaf buffer unterminated.

Reply:
reworked with `METER_BUFFER_APPEND_CHR`